<launch>
	<arg name="pkg_name" value="led_detection"/>
	<arg name="src_name" value="led_detector_node.py"/>
	<arg name="node_name" value="led_detector_node"/>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>
	<arg name="continuous" default="True"/> <!-- manual trigger required if false [interactive mode] -->

	<arg name="protocol_node_name" value="led_emitter_node"/>
	<arg name="protocol_file_name" value="LED_protocol"/>

    <!-- Run detector node -->
    <group ns="$(arg veh)">
        <node  pkg="$(arg pkg_name)" type="$(arg src_name)" name="$(arg node_name)" output="screen" clear_params="true" required="true">
		    <rosparam command="load" file="$(find led_emitter)/config/$(arg protocol_node_name)/$(arg protocol_file_name).yaml"/>
		    <rosparam command="load" file="$(find led_detection)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
		    <param name="continuous" type="bool" value="$(arg continuous)" />
	</node>

    </group>
</launch>
